sql/updates/reverse-dev.sql

```diff
@@ -2,3 +2,88 @@ DROP VIEW _timescaledb_internal.hypertable_chunk_local_size;
 DROP FUNCTION _timescaledb_internal.relation_size(relation REGCLASS);
 DROP INDEX _timescaledb_catalog.chunk_constraint_dimension_slice_id_idx;
 CREATE INDEX chunk_constraint_chunk_id_dimension_slice_id_idx ON _timescaledb_catalog.chunk_constraint (chunk_id, dimension_slice_id);
+
+DO
+$$
+DECLARE
+    caggs_finalized TEXT;
+    caggs_count INTEGER;
+BEGIN
+    SELECT
+        string_agg(format('%I.%I', user_view_schema, user_view_name), ', '),
+        count(*)
+    INTO
+        caggs_finalized,
+        caggs_count
+    FROM
+        _timescaledb_catalog.continuous_agg
+    WHERE
+        finalized IS TRUE;
+
+    IF caggs_count > 0 THEN
+        RAISE EXCEPTION 'Downgrade is not possible because there are % continuous aggregates using the finalized form: %', caggs_count, caggs_finalized
+            USING HINT = 'Remove the corresponding continuous aggregates manually before downgrading';
+    END IF;
+END;
+$$
+LANGUAGE 'plpgsql';
+
+DROP VIEW IF EXISTS timescaledb_information.hypertables;
+DROP VIEW IF EXISTS timescaledb_information.continuous_aggregates;
+
+ALTER TABLE _timescaledb_catalog.continuous_agg
+    DROP COLUMN finalized;
+
+--
+-- Rebuild the catalog table `_timescaledb_catalog.continuous_agg`
+--
+-- We need to recreate the catalog from scratch because when we drop a column
+-- Postgres mark the `pg_attribute.attisdropped=TRUE` instead of removing it from
+-- the `pg_catalog.pg_attribute` table.
+--
+-- If we downgrade and upgrade the extension without rebuild the catalog table it
+-- will mess with `pg_attribute.attnum` and we will end up with issues when trying
+-- to update data in those catalog tables.
+--
+ALTER EXTENSION timescaledb
+    DROP TABLE _timescaledb_catalog.continuous_agg;
+
+ALTER TABLE _timescaledb_catalog.continuous_aggs_materialization_invalidation_log
+    DROP CONSTRAINT continuous_aggs_materialization_invalid_materialization_id_fkey;
+
+CREATE TEMP TABLE __tmp_ts_catalog_continuous_agg ON COMMIT DROP
```

Comment:
this won't work in certain environments, you can't create temp tables when run under pgextwlist